Update addalt, edit feature and add test cases

limjunxian1 opened the pull request:
This PR will

1. Add the functionality of users being able to add 1 more alternative contact field, namely birthday
2. Update edit command to support editing the 4 alternative contact fields
3. Debug the issue raised as mentioned in #71 
4. Add relevant test cases for `addalt`, `edit` features, as well as `linkedin` and `telegram` class created previously

Closes #71, closes #72, closes #73, closes #74

limjunxian1 committed:
Merge branch 'master' into implement-addalt

limjunxian1 committed:
Merge branch 'master' into implement-addalt

limjunxian1 committed:
Implement addalt for birthday and fix bugs

Current implementation shows success message when adding information
to alternative contact fields which already exists.

This is a bug as any updates to information should be done through
another command edit.

limjunxian1 committed:
Update edit command

Current implementation does not allow to edit the alternative
contact information of a person.

Relevant changes are made so that these fields can be edited.

limjunxian1 committed:
Update Ui to display birthday

limjunxian1 committed:
Remove unnecessary code and improve addalt implementation

There are some unused methods in AddAltCommand class.

Moreover, the string pattern in Linkedin and Telegram class enables the
acceptance of weird values.

To reduce clutter, code not needed is removed and the VALIDATION_REGEX
is changed to simulate like the real world for Linkedin and Telegram.

limjunxian1 committed:
Add tests to addalt, edit, person, telegram and linkedin

limjunxian1 committed:
Fix style issues

limjunxian1 committed:
Add more test cases to improve coverage

conradsoon commented on src/main/java/seedu/address/logic/commands/EditCommand.java:
test

conradsoon commented on src/main/java/seedu/address/logic/commands/EditCommand.java:
```diff
@@ -53,7 +53,11 @@ public class EditCommand extends Command {
 
```
Remember to tweak the MESSAGE_USAGE as well since now edit can edit birthdays & other fields.

limjunxian1 replied:
agreed! thanks for pointing that out! fixed in e7540ca.

conradsoon commented on src/main/java/seedu/address/model/person/Person.java:
```diff
@@ -178,6 +176,41 @@ public boolean equals(Object other) {
                 && tags.equals(otherPerson.tags);
```
Found bug where edit command fails and claims fields are same even though I'm setting a different Telegram field. Solution is to add equals checks for telegram, linkedin, birthday fields here, so that persons are not equal when these fields are different.

conradsoon replied:
Should add unit tests for this by the way.

limjunxian1 replied:
Thanks for pointing it out! Didn't really notice on that. Updated the `equals` method in `Person` class accordingly with the relevant test cases as suggested. Fixed in d9bc1b7.

limjunxian1 committed:
Add more test cases to improve coverage and remove unused method

conradsoon reviewed (approved):
Thanks for working on this feature! Generally well-done & with good unit testing.

I found a slight bug when it comes to using the edit command.
edit tg/@alice fails with same field error. even though tg was previously set to @bob for instance. I think I've found the source of the bug, could you double-check?

Other than that, LGTM, feel free to merge whenever.

limjunxian1 committed:
Fix edit command bug

Currently, edit fails even though an attribute of Person is present and
user likes to change to another value.

The equals method of Person is not updated accordingly; it does not
take into account other alternative contact information added.

Equals method in Person is updated and test cases are added to
maintain code coverage.

limjunxian1 committed:
Update edit command usage

limjunxian1 committed:
Fix style issues

limjunxian1 commented:
Thanks for the being so spot on @conradsoon. I've addressed everything as mentioned. Will proceed to merge this. If there's anything I missed, I'll cover it in a future PR.
